float64 vx
float64 vy
float64 vz
float64 theta1
float64 theta2

---
float64 vtheta1
float64 vtheta2
float64 vd3
